cloud/services/loadbalancers/loadbalancers.go

```diff
@@ -0,0 +1,228 @@
+			if err != nil {
+				return errors.Wrap(err, "failed to get subnet")
+			}
+			s.Scope.V(2).Info("successfully got subnet", "subnet", lbSpec.SubnetName)
+			frontIPConfig = network.FrontendIPConfigurationPropertiesFormat{
+				PrivateIPAllocationMethod: network.Static,
+				Subnet:                    &subnet,
+				PrivateIPAddress:          to.StringPtr(privateIP),
+			}
+		} else {
+			s.Scope.V(2).Info("getting public ip", "public ip", lbSpec.PublicIPName)
+			publicIP, err := s.PublicIPsClient.Get(ctx, s.Scope.ResourceGroup(), lbSpec.PublicIPName)
+			if err != nil && azure.ResourceNotFound(err) {
+				return errors.Wrap(err, fmt.Sprintf("public ip %s not found in RG %s", lbSpec.PublicIPName, s.Scope.ResourceGroup()))
+			} else if err != nil {
+				return errors.Wrap(err, "failed to look for existing public IP")
+			}
+			s.Scope.V(2).Info("successfully got public ip", "public ip", lbSpec.PublicIPName)
+			frontIPConfig = network.FrontendIPConfigurationPropertiesFormat{
+				PrivateIPAllocationMethod: network.Dynamic,
+				PublicIPAddress:           &publicIP,
+			}
+		}
+
+		lb := network.LoadBalancer{
+			Sku:      &network.LoadBalancerSku{Name: network.LoadBalancerSkuNameStandard},
+			Location: to.StringPtr(s.Scope.Location()),
+			Tags: converters.TagsToMap(infrav1.Build(infrav1.BuildParams{
+				ClusterName: s.Scope.ClusterName(),
+				Lifecycle:   infrav1.ResourceLifecycleOwned,
+				Role:        to.StringPtr(lbSpec.Role),
+				Additional:  s.Scope.AdditionalTags(),
+			})),
+			LoadBalancerPropertiesFormat: &network.LoadBalancerPropertiesFormat{
+				FrontendIPConfigurations: &[]network.FrontendIPConfiguration{
+					{
+						Name:                                    &frontEndIPConfigName,
+						FrontendIPConfigurationPropertiesFormat: &frontIPConfig,
+					},
+				},
+				BackendAddressPools: &[]network.BackendAddressPool{
+					{
+						Name: &backEndAddressPoolName,
+					},
+				},
+				OutboundRules: &[]network.OutboundRule{
+					{
+						Name: to.StringPtr("OutboundNATAllProtocols"),
+						OutboundRulePropertiesFormat: &network.OutboundRulePropertiesFormat{
+							Protocol:             network.LoadBalancerOutboundRuleProtocolAll,
+							IdleTimeoutInMinutes: to.Int32Ptr(4),
+							FrontendIPConfigurations: &[]network.SubResource{
+								{
+									ID: to.StringPtr(fmt.Sprintf("/%s/%s/frontendIPConfigurations/%s", idPrefix, lbSpec.Name, frontEndIPConfigName)),
+								},
+							},
+							BackendAddressPool: &network.SubResource{
+								ID: to.StringPtr(fmt.Sprintf("/%s/%s/backendAddressPools/%s", idPrefix, lbSpec.Name, backEndAddressPoolName)),
+							},
+						},
+					},
+				},
+			},
+		}
+
+		if lbSpec.Role == infrav1.APIServerRole || lbSpec.Role == infrav1.InternalRole {
+			probeName := "HTTPSProbe"
+			lb.LoadBalancerPropertiesFormat.Probes = &[]network.Probe{
+				{
+					Name: to.StringPtr(probeName),
+					ProbePropertiesFormat: &network.ProbePropertiesFormat{
```

Comment:
this is the only thing I changed that's not strictly a refactor... The probe for the public LB (api server role) was [still using TCP ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/cloud/services/publicloadbalancers/publicloadbalancers.go#L113), I changed it to HTTPS to match https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/316a2a694bef75158af76a99084734517bc0f457 since we're using standard LBs everywhere, and it made it simpler to not have a separate probe if/else per role. Tested functional behavior is the same and this change is backwards compatible.

Happy to revert it if reviewers think a separate PR would be more appropriate and we'd rather keep this one as a strict refactor.

Reply:
Thanks for heads up, was about to comment on the if .. else part. But this comment made it clear as to this is just a copy paste rather than a complete refactor.